### PR TITLE
Doc 49 overview update

### PR DESCRIPTION
**Pull request by Rachel-Reverie**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Added the Setup: Overview page for the Universal Map update.
- Added Step 1: Setup to the sidebar along with related How To guides.

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [X] I have made corresponding changes to the documentation

Thank you for submitting!

### CODE REVIEW

File: docs/guides/setup/setup_overview.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import UniversalMap from '/docs/images/universal_map/_universal_map.mdx';
+
+<!--Use 'inactive' or 'active' to indicate which Universal Map steps this term has a use case within.-->
+
+<UniversalMap setup='active' connect='inactive' create='inactive' validate='inactive'/>
+
+<!-- Only keep one of the 'To best understand this document' lines.  For processes like the Universal Map steps, use the first one.  For processes like the Architecture Reviews, use the second one. -->
+
+:::note Prerequisites
+- To best understand this document, you must have completed Step 1: Setup of the [Getting Started tutorial](../../tutorials/getting_started/initialize_a_data_context.md).
```
Comment (abegong) [inline docs/guides/setup/setup_overview.md]: This feels a bit wordy, IMO. Potential alternatives:
"This document assume that you have already..."
"We recommend completing Step 1: Setup"

Also, I would have thought that we want people to complete the whole tutorial before coming to this page. No?

File: docs/guides/setup/setup_overview.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+:::
+	
+Getting started with Great Expectations is quick and easy.  Once you have completed your setup, you will have access to all of the features of Great Expectations from a single entry point: Your Data Context.  You will also have your Metadata Stores and Data Docs configured in the manner most suitable for your project's purposes.
```
Comment (abegong) [inline docs/guides/setup/setup_overview.md]: Style Guide question: "Metadata Stores" or "Stores"?

We don't actually have an abstraction called MetadataStore, but we do have Store. OTOH, "Metadata Store" is a good way to let people know that we're talking about a specific technical thing.

Comment (abegong) [reply docs/guides/setup/setup_overview.md]: A decision on this would be really helpful. I run into it all the time in our docs.

File: docs/guides/setup/setup_overview.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+## The Setup process
+
+<!-- Brief outline of what the process entails.  -->
+
+Setup entails ensuring your system is prepared to run Great Expectations, installing Great Expectations itself, initializing your deployment, and doing some optional configuration of things like Data Docs and Datastores.  We'll look at each of these things in sequence.
```
Comment (abegong) [inline docs/guides/setup/setup_overview.md]:
```suggestion
Setup entails ensuring your system is prepared to run Great Expectations, installing Great Expectations itself, and initializing your deployment. Optionally, you can also tweak the configuration of some components, such as Metadata Stores and Data Docs. We'll look at each of these things in sequence.

Note: configuration of Datasources, Expectation Suites, and Checkpoints will be handled separately. We consider those to be configuration of components after your main Great Expectations deployment is set up.
```

File: docs/guides/setup/setup_overview.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+<!-- The following subsections should be repeated as necessary.  They should give a high level map of the things that need to be done or optionally can be done in this process, preferably in the order that they should be addressed (assuming there is one). If the process crosses multiple steps of the Universal Map, use the <SetupHeader> <ConnectHeader> <CreateHeader> and <ValidateHeader> tags to indicate which Universal Map step the subsections fall under. -->
+
+### 1. System Dependencies
+
+The first thing to take care of is making sure your work environment has the utilities you need to install and run Great Expectations.  These include a working Python install (version 3.8 or greater), the ability to pip install Python packages, an internet connection, and a browser so that you can use Jupyter notebooks.  Best practices are to use a virtual environment for your project's workspace.
```
Comment (abegong) [inline docs/guides/setup/setup_overview.md]: Is version 3.8 our minimum? If it's not already, this should be publicly documented.

File: docs/guides/setup/setup_overview.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+Running this command in an environment configured to accept Python pip install commands will handle the entire installation process for Great Expectations and its dependencies.
+
+## 3. Initialize a Data Context
+
+Your Data Context contains the entirety of your Great Expectations project and provides the entry point for all of the primary methods you will use to configure and interact with Great Expectations.  That's why the first thing you'll do once you've installed Great Expectations will be to initialize your Data Context.  And this is another one-line command.  Simply go to the root folder for your project and execute:
+
+```markdown title="Terminal command:"
+great_expectations --v3-api init
```
Comment (abegong) [inline docs/guides/setup/setup_overview.md]:
```suggestion
great_expectations init
```

Starting with the 0.14 release in January, v3 will be the default.

File: docs/guides/setup/setup_overview.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+```markup title="Terminal command:"
+pip install great_expectations
+```
+
+Running this command in an environment configured to accept Python pip install commands will handle the entire installation process for Great Expectations and its dependencies.
+
+## 3. Initialize a Data Context
+
+Your Data Context contains the entirety of your Great Expectations project and provides the entry point for all of the primary methods you will use to configure and interact with Great Expectations.  That's why the first thing you'll do once you've installed Great Expectations will be to initialize your Data Context.  And this is another one-line command.  Simply go to the root folder for your project and execute:
```
Comment (abegong) [inline docs/guides/setup/setup_overview.md]: Another style guide question: "Project" or "Deployment"?

Either is fine. I lean towards "deployment," but it's not my decision. Let's pick one and standardize on it.

Comment (abegong) [reply docs/guides/setup/setup_overview.md]:
```suggestion
Your Data Context contains the entirety of your Great Expectations project and provides the entry point for all of the primary methods you will use to configure and interact with Great Expectations.  That's why the first thing you'll do once you've installed Great Expectations will be to initialize your Data Context.  This is another one-line command.  Simply go to the root folder for your project and execute:
```

File: docs/guides/setup/setup_overview.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+```markdown title="Terminal command:"
+great_expectations --v3-api init
+```
+
+Running this command will initialize your Data Context in the directory that the command is run from.  It will create the folder structure a Data Context requires to organize your project.
+
+## 4. Optional Configurations
```
Comment (abegong) [inline docs/guides/setup/setup_overview.md]: This would be a great section for a diagram to show how the pieces fit together.

Here's a rough draft, with some draft notes. I don't love these---feel like I'm probably another conversation or two away from nailing down the groupings that I could really stand behind.

Components of a Data Context

### What are the top-level components within a Great Expectations DataContext?

There are 9 top-level components in a DataContext configuration.

Four of these components fit within the three required steps for setting up Great Expectations.

	* Connect to data: DataSources
	* Create and edit Expectations:
		* ExpectationStore
		* ProfilerStore
	* Validate data : CheckpointStore

Three of the components are intended to be managed internally by Great Expectations.

	* DataDocs
	* ValidationStore
	* EvaluationParameterStore (aka MetricStore)

Two are Advanced:
	
	* plugins
	* config_variables_path (Maybe rename this to SecretsStore?)

**Review by abegong — CHANGES_REQUESTED** [review]

File: docs/guides/setup/setup_overview.md (hunk 4 above)
Comment (Rachel-Reverie) [reply docs/guides/setup/setup_overview.md]: I'm not certain if this is our minimum, but it is the version our on boarding guidance tells us to install.

File: docs/guides/setup/setup_overview.md (hunk 1 above)
Comment (Rachel-Reverie) [reply docs/guides/setup/setup_overview.md]: Ideally, yes.  But it's also possible that people will skip the tutorial and come straight to this section, or (hopefully the more likely scenario) they may click a Technical Term for more information, then follow the "Setup" link back to this page.  In which case having a link at the top of this page back to the step in the tutorial that they were on would be useful.

Also, in general I want our prerequisites to not contain anything extraneous.  If you don't actually need to have done something to follow a guide or understand a documentation page, I don't want us to be telling people they do.  We want them to have completed the tutorial before they follow any how to guides so that they have the additional context, yes.  But if they just want to read about a given subject without doing anything (as they would on the overview or core concept/definitions pages -- which I don't intend to put prerequisites on at all) I don't think we need to require that.  If you disagree with my reasoning for these non-instruction pages, please let me know so we can evaluate and decide.

I will change the prerequisite note to "Completing Step 1: Setup in the Getting Started tutorial is recommended."

File: docs/guides/setup/setup_overview.md (hunk 2 above)
Comment (Rachel-Reverie) [reply docs/guides/setup/setup_overview.md]: I believe in our discussion over the Glossary terms, we changed it from "Metadata Stores" to "Stores" on the premise that "Stores" is a very clearly broad and general term, while attaching anything else to implies a subcategory of Store.

If I'm misremembering and we didn't explicitly say that, I'll go ahead and do so now: We should use "Store" because the way we have named all other Stores could result in confusion -- eg, it would be more intuitive to understand that an Expectation Store was a subcategory of Metadata Store if it were instead named "Expectation Metadata Store."  On the other hand, "Expectation Store" is clearly a specification of "Store".  But it would be understandable that if someone saw "Metadata Store" and "Expectation Store" side by side they thought they were two entirely different things -- one a Store containing Metadata, and the other a Store containing Expectations.

I'll update the Setup: Overview and related documents accordingly.

File: docs/guides/setup/setup_overview.md (hunk 7 above)
Comment (Rachel-Reverie) [reply docs/guides/setup/setup_overview.md]: I'm not sure if we want to insert this as a diagram here, which will make this section seem much longer than the others.  It might be better to have be a reference page that we link to from the Wrapping up section (which we could also have the other Overview pages link to, given that the elements encompassed in this diagram will be relevant across steps).  Something like: "For more information on how Great Expectations elements are organized within a Data Context, see our reference diagram for Great Expectations: Hierarchy of Components" while other Overview pages would have something more like: "For more information on how Profilers and Validators fit into the Great Expectations architecture, see our reference diagram..."

Thoughts?

**Comment by abegong** [comment]
The new index pages look good, and will definitely be useful. (Probably a bit of a pain to maintain, unless you've scripted the process for building them...)

One thought: might the go better at the end of their sections, rather than the beginning? That would be more like the index to a book. At the beginning, I find myself vaguely wanting an introduction or explanatory text...

I can see it both ways---just wanted to raise the idea for consideration before locking into a specific implementation.

File: docs/guides/setup/setup_overview.md (hunk 7 above)
Comment (abegong) [reply docs/guides/setup/setup_overview.md]: I figure it'll go in one of those two places.

I think it depends on whether you want the high-level "How this thing relates to other things" to live at the "Overview" level or the "Reference" level. FWIW, I'm imagining that each of the 4 steps has a diagram as an important part of its explanation.

IMO, one diagram at the Overview level makes sense, because
1. the Getting Started tutorial is primarily a list of steps and this would let Overview be something more than a list of steps, and
2. in most cases, there'll be another, more intricate diagram that can live in Reference.

I'll be the first to admit that these are pretty soft considerations.

File: docs/guides/setup/setup_overview.md (hunk 4 above)
Comment (abegong) [reply docs/guides/setup/setup_overview.md]: Understood. Don and/or James should be able to give you an official version.

File: docs/guides/setup/setup_overview.md (hunk 7 above)
Comment (Rachel-Reverie) [reply docs/guides/setup/setup_overview.md]: I think that makes sense.  I may tweak exactly where in the overview documentation the diagram exists -- I'd like to maintain a certain level of uniformity between documents of the same type, so if there's a place that makes sense to have the 'overview diagram' in all four overview pages, I'd like to standardize that.

File: docs/guides/setup/setup_overview.md (hunk 4 above)
Comment (Rachel-Reverie) [reply docs/guides/setup/setup_overview.md]: I've updated this according to feedback from Don.

File: docs/guides/setup/setup_overview.md (hunk 7 above)
Comment (Rachel-Reverie) [reply docs/guides/setup/setup_overview.md]: Per our meeting, I'm going to mark this conversation as resolved and move on with merging this PR.  We'll add illustrations to each of the overview pages in separate PRs so that hashing out the exact message and content of the illustrations doesn't put a hold the initial content updates to our documentation.

**Comment by Rachel-Reverie** [comment]
I've updated the placement of the index pages per @abegong 's suggestion.  Currently they are manually generated, but I expect that it will be straightforward to write a script that will generate them out of `sidebars.js`.

**Review by donaldheppner — APPROVED** [review]

**Review by abegong — APPROVED** [review]
❤️ ❤️ ❤️ ❤️ ❤️ 

(I haven't re-reviewed all text, but we've discussed enough that I'm very comfortable with this.)
